Error on operating systems using DNF5 if wazuh-agent is not stopped before uninstalling it

## Description

During issue https://github.com/wazuh/wazuh/issues/35761 it was discovered that if a stop command is not added before uninstalling in Fedora 44, an error is detected.

```console
    ============================= test session starts ==============================
    platform linux -- Python 3.10.12, pytest-8.3.2, pluggy-1.6.0 -- /home/rafa/Repositories/wazuh-qa-automation/env/bin/python3
    cachedir: .pytest_cache
    rootdir: /home/rafa/Repositories/wazuh-qa-automation
    collecting ... collected 5 items
  
    ../../../tests/test_functional/test_system/test_deployability/tests/test_agent/test_uninstall.py::test_uninstall [32m[2026-05-15 08:48:30] [utils.py] [INFO] TESTING: Checking connection to fedora-44
    [32m[2026-05-15 08:48:32] [utils.py] [INFO] TESTING: Connection established successfully in fedora-44
    [32m[2026-05-15 08:48:42] [agent.py] [INFO] TESTING: Uninstalling Agent in fedora-44
    PASSED
    ../../../tests/test_functional/test_system/test_deployability/tests/test_agent/test_uninstall.py::test_agent_uninstalled_directory PASSED
    ../../../tests/test_functional/test_system/test_deployability/tests/test_agent/test_uninstall.py::test_service [31m[2026-05-15 08:49:08] [test_uninstall.py] [ERROR] TESTING: agent-linux-fedora-44-amd64 is not inactive by command
    FAILED
    ../../../tests/test_functional/test_system/test_deployability/tests/test_agent/test_uninstall.py::test_port PASSED
    ../../../tests/test_functional/test_system/test_deployability/tests/test_agent/test_uninstall.py::test_processes PASSED
  
    =================================== FAILURES ===================================
    _________________________________ test_service _________________________________
  
    wazuh_params = {'agents': {'agent-linux-fedora-44-amd64': '/tmp/dtt-poc/agent-linux-fedora-44-amd64/inventory.yaml'}, 'dependencies': '{}', 'live': 'True', 'master': '/tmp/dtt-poc/manager-linux-ubuntu-22.04-amd64/inventory.yaml', ...}
  
        def test_service(wazuh_params):
            wazuh_api = WazuhAPI(wazuh_params['master'])
            for agent_names, agent_params in wazuh_params['agents'].items():
    >           assert not GeneralComponentActions.is_component_active(agent_params, 'wazuh-agent', silent=True), logger.error(f'{agent_names} is not inactive by command')
    E           AssertionError: None
    E           assert not True
    E            +  where True = <function GeneralComponentActions.is_component_active at 0x7f2a05a2cc10>('/tmp/dtt-poc/agent-linux-fedora-44-amd64/inventory.yaml', 'wazuh-agent', silent=True)
    E            +    where <function GeneralComponentActions.is_component_active at 0x7f2a05a2cc10> = GeneralComponentActions.is_component_active
  
    ../../../tests/test_functional/test_system/test_deployability/tests/test_agent/test_uninstall.py:92: AssertionError
```

I tried adding `systemctl stop wazuh-agent` before using `yum remove wazuh-agent -y` and the error was fixed. It's necessary to investigate further why this is happening and whether it's necessary to add the command to stop the agent. If so, the documentation needs to be updated. From my initial investigation, it appears to be related to `DNF5`. More information:

- https://github.com/wazuh/wazuh/issues/35761#issuecomment-4457483876
- https://github.com/wazuh/wazuh-qa-automation/pull/7227#discussion_r3316392297

## Tasks

- [ ] Investigate why it fails in Fedora 44 when the agent is not stopped beforehand
- [ ] Update the documentation if it is necessary to stop the agent before uninstalling it.

--- a/src/init/templates/wazuh-agent.service
+++ b/src/init/templates/wazuh-agent.service
@@ -12,7 +12,6 @@ ExecStop=/usr/bin/env WAZUH_HOME_TMP/bin/wazuh-control stop
 ExecReload=/usr/bin/env WAZUH_HOME_TMP/bin/wazuh-control reload
 
 KillMode=process
-RemainAfterExit=yes
 
 [Install]
 WantedBy=multi-user.target

